src/Microsoft.DotNet.Build.Tasks/GenerateDepsJson.cs

```diff
@@ -110,52 +110,56 @@ public override bool Execute()
                 }
             }
 
-            targetsSection.Remove(runtimeTarget);
-            newDepsJson.Remove("libraries");
+            string platformPackageName = "";
+            foreach (var package in targetsSection[runtimeTarget].Children())
+            {
+                JProperty packageProperty = package as JProperty;
+                if (packageProperty != null && packageProperty.Name.Contains($"runtime.{rid}.Microsoft.NETCore.App"))
+                {
+                    platformPackageName = packageProperty.Name;
+                    break;
+                }
+            }
+
+            if (string.IsNullOrEmpty(platformPackageName))
+                Log.LogError($"Unable to find platform package that matches runtime.{rid}.Microsoft.NETCore.App when generating deps.json file.");
```

Comment:
I take it there was too much work to make both scenarios work?

Reply:
I didn't try given that I was moving all of the runtimeconfig.jsons to point to 2.0 so that meant that the old scenario wouldn't work anymore. Now that I'll try to make that happen on a script, I'll make both scenarios work here. If it's not easy to autodetect which deps.json should be generated, I could always fallback to an extra parameter to indicate which generation of deps.json should be built.